src/views/components/survey_components/question_time_setting.vue

```diff
@@ -175,27 +180,36 @@ export default {
 
             // Set the default survey start and end time according to the JSON data
             // sent back
-            let [date, time] = fetchedJSON.DATA.start_time.split(" ");
+            let [date, time] = fetchedJSON.DATA[0].start_time.split(" ");
             this.start_time = date + "T" + time;
-            [date, time] = fetchedJSON.DATA.end_time.split(" ");
+            [date, time] = fetchedJSON.DATA[0].end_time.split(" ");
             this.end_time = date + "T" + time;
-        }
-        // Fill in the question_id_array
-        for (let i = 0; i < this.questions.length; i++) {
-            this.question_id_array.push(this.questions[i].question_id);
-        }
-        // Fetch all the questions back from the API
-        fetched = await fetch("get_info.php?what=questions");
-        fetchedJSON = await fetched.json();
-        // Determine which question objects to put into the backup_questions list
-        this.backup_questions = fetchedJSON.DATA.filter(this.question_filter);
-    },
-    methods: {
+
+            // Fill in the question_id_array
+            for (let i = 0; i < this.questions.length; i++) {
+                this.question_id_array.push(this.questions[i].question_id);
+            }
+
+            // Fetch all the questions back from the API
+            fetch("get_info.php?what=questions")
+                .then(res => res.json())
+                .then(
+                    data =>
+                        (this.backup_questions = data.DATA.filter(
+                            this.question_filter
+                        ))
```

Comment:
This is really weird...Change this function to use `{...}` syntax. Right now you're using an implicit return. In general, if the code formatting looks strange, you're probably not writing good code...

Reply:
Do you mean use `{}` and have an explicit return? Or do you mean use the spread operator?

Reply:
You don't need an explicit return. Replace `()` with `{}`